src/App/VPSScan/Scan/Core.hs

```diff
@@ -37,7 +43,29 @@ instance FromJSON SherlockInfo where
   parseJSON = withObject "SherlockInfo" $ \obj -> do
     auth <- obj .: "auth"
     SherlockInfo <$> obj .: "url" <*> auth .: "clientToken" <*> auth .: "clientId" <*> obj .: "orgId"
-    
+
+data CoreFilterExpressions = CoreFilterExpressions
+  { filters :: [Text]
+  }
+  deriving (Eq, Ord, Show)
+
+instance FromJSON CoreFilterExpressions where
+  parseJSON = withObject "CoreFilterExpressions" $ \obj -> CoreFilterExpressions <$> obj .: "filters"
+
+instance ToJSON CoreFilterExpressions where
+  toJSON (CoreFilterExpressions filters) = object ["filters" .= filters]
+
+toFilterExpressions :: CoreFilterExpressions -> FilterExpressions
+toFilterExpressions coreFilterExpressions = do
+  let encoded = decodeUtf8 $ BSL.toStrict $ encode (filters coreFilterExpressions)
+  FilterExpressions encoded
+
+toCoreFilterExpressions :: FilterExpressions -> Either String CoreFilterExpressions
+toCoreFilterExpressions filterBlob = do
+  let blob = "{\"filters\":" <> (unFilterExpressions filterBlob) <> "}"
+  eitherDecode $ BSL.fromStrict $ (encodeUtf8 blob)
```

Comment:
This is easily the grossest part of this PR to me. I am sure there is a better way to reuse the JSON parsing functionality than this, but wasn't able to find how to do so. Would love guidance here.

Reply:
I think this forces us to care about the actual list of strings passed to us in `--ignore-file-regex`. Rather than parsing it as `Text`, we should actually decode the json:
```hs
newtype FilterExpressions = FilterExpressions { unFilterExpressions :: [Text] }
```

and adjusting the `filterOpt` in `App.VPSScan.Main`:
```hs
-- this is like `strOption` with a slightly different type signature
jsonOption :: FromJSON a => Mod OptionFields a -> Parser a
jsonOption = option (eitherReader (eitherDecode . fromString))

filterOpt :: Parser FilterExpressions
filterOpt = FilterExpressions <$> jsonOption (long "ignore-file-regex" <> short 'i' <> metavar "REGEXPS" <> help "JSON encoded array of regular expressions used to filter scanned paths" <> value "[]")
```

(with `eitherDecode` from `Data.Aeson` and `fromString` from `Data.String`)

this should simplify things across this PR

Reply:
ooh, ok! Thanks.